File: DSA/GEHNA_KHANNA_590013140/day34question3.c
#include <stdio.h>
int main() 
{
    int n;
    printf("Enter a number: ");
    scanf("%d", &n);
    int left = 31; 
    while (((n >> left) & 1) == 0 && left > 0) 
    {
        left--; 
    }
    int right = 0; 
    int isPalindrome = 1;
    while (left > right) 
    {
        int leftBit = (n >> left) & 1;
        int rightBit = (n >> right) & 1;
        if (leftBit != rightBit) 
        {
            isPalindrome = 0; 
            break;
        }
        left--;
        right++;
    }
    if (isPalindrome)
        printf("Yes\n");
    else
        printf("No\n");

    return 0;
}

File: DSA/NiharikaChand_590017697/Day34Question3.java
import java.util.*;
class Day34Question3
{
    public static String isPalin(int n) 
    {
        int msb = -1;
        for (int i = 31; i >= 0; i--) 
        {
            if (((n >> i) & 1) == 1) 
            {
                msb = i;
                break;
            }
        }

        int left = msb; 
        int right = 0;  

        while (left > right) 
        {
            int leftBit = (n >> left) & 1;
            int rightBit = (n >> right) & 1;

            if (leftBit != rightBit) 
            {
                return "No";
            }

            left--;
            right++;
        }
        return "Yes";
    }

    public static void main(String args[]) 
    {
        Scanner sc = new Scanner(System.in);
        System.out.print("Enter a number: ");
        int n = sc.nextInt();

        String result = isPalin(n);
        System.out.println("Output: " + result);
    }
}
